sympy/core/basic.py

```diff
@@ -1216,7 +1216,8 @@ def _has(self, pattern):
             return any(f.func == pattern or f == pattern
             for f in self.atoms(Function, UndefinedFunction))
 
-        pattern = _sympify(pattern)
+        if not (isinstance(pattern, type) and issubclass(pattern, Basic)):
+            pattern = _sympify(pattern)
         if isinstance(pattern, BasicMeta):
```

Comment:
This first if is essentially mutually exclusive with the second, isn't it? So perhaps it should be changed to an `else`

Reply:
Oh, good point. Actually I should have just swapped the order of these.

Reply:
I've just moved the call to `_sympify` after the check for `BasicMeta`.